core/src/test/java/com/opensymphony/xwork2/interceptor/CSRFPreventionInterceptorTest.java

```diff
@@ -0,0 +1,152 @@
+package com.opensymphony.xwork2.interceptor;
+
+
+import com.opensymphony.xwork2.ActionContext;
+import com.opensymphony.xwork2.XWorkTestCase;
+import com.opensymphony.xwork2.mock.MockActionInvocation;
+import org.apache.struts2.ServletActionContext;
+import org.springframework.mock.web.MockHttpServletRequest;
+
+import java.util.Arrays;
+import java.util.List;
+
+// TODO Vary headers tests
+
+public class CSRFPreventionInterceptorTest extends XWorkTestCase {
```

Comment:
test failures will be much easier to read if we use `assertTrue(message, condition)` instead of the pure `assert`. That way the failure will read **expected Fetch Metadata to reject this request** rather than **expected false to be true**